File: app/src/main/java/kr/co/company/capstone/service/LikeService.java
package kr.co.company.capstone.service;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.PATCH;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface LikeService {
    // 좋아요
    // /goals/{goalId}/posts/{postId}/like
    @POST("/goals/{goalId}/posts/{postId}/like")
    Call<Void> like(@Path("goalId") long  goalId ,@Path("postId")long postId);

    // 좋아요 취소
    @DELETE("/goals/{goalId}/posts/{postId}/unlike")
    Call<Void> unlike(@Path("goalId") long  goalId, @Path("postId")long postId);

    static LikeService getService(){
        return RetrofitBuilder.INSTANCE.getRetrofit().create(LikeService.class);
    }
}

File: app/src/main/java/kr/co/company/capstone/service/LikeService.kt
package kr.co.company.capstone.service

import retrofit2.Call
import retrofit2.http.DELETE
import retrofit2.http.POST
import retrofit2.http.Path

interface LikeService {
    // 좋아요
    @POST("/goals/{goalId}/posts/{postId}/like")
    fun like(@Path("goalId") goalId: Long, @Path("postId") postId: Long): Call<Void>

    // 좋아요 취소
    @DELETE("/goals/{goalId}/posts/{postId}/unlike")
    fun unlike(@Path("goalId") goalId: Long, @Path("postId") postId: Long): Call<Void>

    companion object {

        fun getService() : LikeService {
            return RetrofitBuilder.retrofit.create(LikeService::class.java)
        }
    }
}
